docs/operations/batch-processing.md

```diff
@@ -0,0 +1,132 @@
+# Batch Processing
+
+In order to answer queries, JanusGraph has to perform queries
+against the storage backend.
+In general, there are two ways of doing this:
+
+- Once data from the backend is needed, execute a backend
+  query and continue with the result.
+- Maintain a list of what data is needed.
+  Once the list reaches a certain size, execute a single backend
+  query to fetch all of it at once.
+  
+The first option tends to be more responsive because the query
+can emit the first results very early without waiting for larger
+batches of queries to complete.
+This is also the option that JanusGraph uses by default.
+The second option can be configured in multiple ways which are
+explained below.
+
+## Configuration Options
+
+### No Batch Processing
+This is the default configuration of JanusGraph.
+In terms of graph traversals, the execution of queries is
+loosely coupled to the principle of Depth-First-Search.
+
+#### Use this configuration in use cases where for example ...
+- ... each query only accesses few vertices of the graph.
+- ... your application does not need the _full_ result set
+  immediately but rather requires a low latency for the first
+  results to arrive.
+
+#### Possible limitations
+- Traversing large neighborhoods can make the query slow.
+
+#### Steps to explicitly configure this option:
+- Ensure `query.batch` is set to `false`
+
+### Unrestricted Batch Processing
+
+Using this configuration, each step which traverses the Graph
+starting from a vertex (so e.g. `in()`, `outE()` and `values()`
+but not `inV()` or `otherV()`) becomes a blocking operator
+which means that it produces no results until all the results
+of the previous step are known.
+Only then, a single backend query is executed and the results
+are passed to the next step.
+This way of execution can be thought of as a
+Breadth-First-Search.
+
+#### Use this configuration in use cases where for example ...
+- ... your queries are likely to access multiple vertices in
+  each step.
+- ... there is a significant network latency between JanusGraph
+  and the storage backend.
+
+#### Possible limitations
+- Limits are only applied very late, so there might be an
+  unnecessary overhead.
+- Performing very large backend queries could stress the
+  storage backend.
+
+#### Steps to explicitly configure this option:
+- Ensure `query.batch` is set to `true`
+- Ensure `query.limit-batch-size` is set to `false`
```

Comment:
I think I don't quite get what you mean here. If `query.limit-batch-size` is set to false, Janusgraph should behave just the same as before. The only difference being that now, the `JanusGraphMultiQueryStep` in front of the next `MultiQueriable` accumulates all previous results.
In this case, the sizes of in-between `barrier()` steps will not affect the amount or constellation of fired backend queries. That behavior is equal to the situation before this PR. If you wish, I can of course clarify that.

Reply:
Right, I shouldn't have said: "after this PR". I am interested in the current behavior from a user standpoint since we don't have a doc for that. Now that you are adding this documentation, it would be great if you could clarify that.